near/omni-types/src/lib.rs

```diff
@@ -37,7 +38,38 @@ impl FromStr for H160 {
 
 impl fmt::Display for H160 {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "0x{}", hex::encode(self.0))
+        let hex_addr = hex::encode(self.0);
+
+        let mut hasher = Keccak256::new();
+        hasher.update(hex_addr.as_bytes());
+        let hash = hasher.finalize();
+
+        let mut result = String::with_capacity(40);
+
+        for (i, c) in hex_addr.chars().enumerate() {
+            let hash_byte = hash[i / 2];
+
+            let hash_nibble = if i % 2 == 0 {
+                (hash_byte >> 4) & 0xF
+            } else {
+                hash_byte & 0xF
+            };
+
+            let c = match c {
+                'a'..='f' => {
+                    if hash_nibble >= 8 {
+                        c.to_ascii_uppercase()
+                    } else {
+                        c
+                    }
+                }
+                _ => c,
+            };
+
+            result.push(c);
+        }
+
+        write!(f, "0x{}", result)
```

Comment:
I would like to avoid these kinds of operations on-chain, especially since using the `alloy_primitives::Keccak256` isn't acceptable; if it is needed just for the relayer, then please implement a separate method `to_eip_55_string` and use the hash function from utils

Reply:
So should I just delete it?

Reply:
You can keep the `to_eip_55_checksum` as a public method if you want, but in the fmt & serd let's use `write!(f, "0x{}", hex::encode(self.0))`

Reply:
Done https://github.com/Near-One/omni-bridge/pull/44/commits/8cedbc8a42fee426aad3b5fcd9530c07c408f659